test/python/quantum_info/states/test_statevector.py

```diff
@@ -335,6 +344,20 @@ def test_tensor(self):
             self.assertEqual(state.dims(), (3, 2))
             assert_allclose(state.data, target)
 
+    def test_inner(self):
+        """Test inner method."""
+        for _ in range(10):
+            seed = np.random.randint(0, np.iinfo(np.int32).max)
+            rng = np.random.default_rng(seed)
+            dim = 2 ** rng.integers(1, 4)
```

Comment:
I get that you're copying the style in the rest of this file, but in general, randomness like this is a bad idea in tests.  If you want to test the method with one to four qubits, then just loop over them - don't randomly sample, because that'll mean that on some test runs, you don't test all four values.

Reply:
Okay. Would it be enough to just loop one time over the four values with randomly generated vectors? Or do we also want to do multiple rounds for each of the four values?